### PR TITLE
[dv,chip,reset_info] Fix tests that read reset_info via difs

### PR DESCRIPTION
**Pull request by matutem**
Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

File: sw/device/tests/sim_dv/lc_ctrl_program_error.c [hunk 1]
```diff
@@ -59,8 +60,7 @@ bool test_main(void) {
       &alert_handler));
 
   // Check if there was a HW reset caused by the escalation.
-  dif_rstmgr_reset_info_bitfield_t rst_info;
-  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  dif_rstmgr_reset_info_bitfield_t rst_info = rstmgr_testutils_reason_get();
   CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
```
Comment (drewmacrae) [inline sw/device/tests/sim_dv/lc_ctrl_program_error.c:64]: This is already cleared. You may want to clear the reset reason here.

File: sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c [hunk 2]
```diff
@@ -45,8 +46,7 @@ bool test_main(void) {
                                               kDifToggleEnabled));
 
   // Check if there was a HW reset caused by the wdog bite.
-  dif_rstmgr_reset_info_bitfield_t rst_info;
-  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  dif_rstmgr_reset_info_bitfield_t rst_info = rstmgr_testutils_reason_get();
   CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
```
Comment (drewmacrae) [inline sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c:50]: This is already cleared by the ROM. You might want to use the testutil to clear the reset reason in retention SRAM

**Review by drewmacrae — APPROVED** [review]
Lgtm thanks for the cleanup

**Review by sriyerg — APPROVED** [review]

**Comment by sriyerg** [comment]
LGTM, barring @drewmacrae's pending comments.

**Comment by tjaychen** [comment]
thanks for fixing the program error test!

File: sw/device/tests/sim_dv/lc_ctrl_program_error.c (hunk 1 above)
Comment (matutem) [reply sw/device/tests/sim_dv/lc_ctrl_program_error.c:64]: If it is already cleared, do you mean there is no need to clear it here?

File: sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c (hunk 2 above)
Comment (matutem) [reply sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c]: Done.

File: sw/device/tests/sim_dv/lc_ctrl_program_error.c (hunk 1 above)
Comment (matutem) [reply sw/device/tests/sim_dv/lc_ctrl_program_error.c]: Done.

**Pull request merged**